server/groundstation.py

```diff
@@ -90,11 +92,37 @@ def uav_thread(self):
             time.sleep(0.1)
 
     def image_thread(self):
-        while True:
-            run = self.image.retreive_images()
-            if run:
-                self.logger.debug("[Image] %s", run)
-            time.sleep(0.1)
+        if self.config["uav"]["images"]["connection"]:  # Initialize a socket connection
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+                s.bind((self.config["uav"]["images"]["host"], self.config["uav"]["images"]["port"]))
+                s.listen()
+                conn, addr = s.accept()
+                with conn:
```

Comment:
It may be better to encapsulate this general loop into another method in `image_handler`.